src/main/java/com/microsoft/azure/AzureVMCloudRetensionStrategy.java

```diff
@@ -105,13 +105,17 @@ public Void call() throws Exception {
                 LOGGER.log(Level.INFO, "AzureVMCloudRetensionStrategy: check: could not terminate or shutdown {0}: {1}",
                         new Object [] { agentNode.getName(), ae });
                 // If we have an exception, set the agent for deletion.  It's unlikely we'll be able to shut it down properly ever.
-                agentNode.getNode().setCleanUpAction(CleanUpAction.DELETE, Messages._Failed_Initial_Shutdown_Or_Delete());
+                AzureVMAgent node = agentNode.getNode();
+                if(node != null)
```

Comment:
IMHO single-line conditionals like this are dangerous as they can hide subtle bugs in logic. I recommend sticking to `if (condition) { }` for all conditionals.


Your call though